packages/safety-suite-api/src/api/models/types.ts

```diff
@@ -944,3 +944,197 @@ export type LostRestrictedInput = {
   estimateLostDays: number;
   estimateRestrictedDays: number;
 };
+
+export type FieldType =
+  | 'boolean'
+  | 'dateTime'
+  | 'enum'
+  | 'group'
+  | 'list'
+  | 'number'
+  | 'text';
+export type RelationType = 'singleModel' | 'multiModel';
+export interface FieldTypes extends Record<FieldType, unknown> {
+  list: string;
+  number: number;
+  enum: string;
+  text: string;
+  boolean: string;
+}
+
+export interface BaseField<Type extends string> {
+  /**
+   * Field name.
+   */
+  name: Alias;
+  /**
+   * Formatted display label.
+   */
+  label?: string;
+  /**
+   * The data type.
+   */
+  type: Type;
+}
+
+export interface BaseFieldOptions<Value extends unknown> {
+  /**
+   * How the value should be formatted. E.g. `currency` for a number type.
+   */
+  formatType?: string;
+  /**
+   * Default value if no value is provided upon creation of the model.
+   */
+  defaultValue?: Value;
+  /**
+   * If the field is included in custom reporting. Can be a boolean or an object to be checked against config.
+   *
+   * Defaults to `true`
+   */
+  onReport?: boolean | {key: string; values: unknown[]};
+  onColumn?: boolean;
+  /**
+   * Indirect modules which use this field.
+   */
+  indirectModules?: Alias[];
+}
+
+export type Required =
+  | boolean
+  | {
+      path: string[];
+      cases: {
+        pattern: string;
+        result: boolean;
+      }[];
+      default?: boolean;
+    };
+
+export type DateType = 'date' | 'time' | 'dateTime';
+
+export interface Field<Type extends FieldType = FieldType>
+  extends BaseField<Type> {
+  options: BaseFieldOptions<FieldTypes[Type]> & {
+    /**
+     * If the field is confidential.
+     */
+    protected: boolean;
+    /**
+     * If the field is required, can be a boolean or an object to be checked against config.
+     */
+    required: Required;
+    sideEffect?: unknown;
+  } & (Type extends 'list'
+      ? {
+          itemType: {
+            name: string;
+            type: FieldType;
+          };
+        }
+      : Type extends 'number'
+      ? {
+          /**
+           * Type of number. E.g. `decimal`
+           */
+          numberType?: string;
```

Comment:
This is only `decimal` today.  The intent was `decimal` and `integer` or something, with `integer` as the default value, but well... why bother ever putting in the default value I guess?